### PR TITLE
REACT-MP-2020 task#4 redux

### PR DESCRIPTION
**Pull request by garnich**
This PR include:

Task 6: (Redux)

For BE side I use MoviesAPI.ReactJS (https://github.com/VarvaraZadnepriak/MoviesAPI.ReactJS).

### CODE REVIEW

File: src/components/header/header.js [hunk 1]
```diff
@@ -60,13 +48,28 @@ const Header = (props) => {
...
-const MemoizedHeader = React.memo(Header);
+const mapStateToProps = (state) => {
+    return {
+        movies: state.movies,
+        id: state.id,
+        showMovieDetails: state.showMovieDetails,
+    }
+  }
+  
+  const mapDispatchToProps = (dispatch) => {
```
Comment (DenisKarlyuk) [inline src/components/header/header.js]: You can use short form `mapDispatchToProps = { hideDetails: movieHideDetails }`

File: src/components/header/header.js [hunk 2]
```diff
@@ -1,45 +1,33 @@
...
 const Header = (props) => {
     const [show, setShow] = useState(false);
-    const [showDetails, setShowDetails] = useState(false);
     
     const handleClose = useCallback(() => setShow(false), []);
     const handleShow = useCallback(() => setShow(true), []);
-    const handleCloseDetails = useCallback(() => setShowDetails(false), []);
-    
-    const movies = useContext(MockDataContext);
+    const handleCloseDetails = useCallback(() => hideDetails(), []);
```
Comment (DenisKarlyuk) [inline src/components/header/header.js]: `showDetails` is used only this component, it's not best way to save it to `redux` store

File: src/components/header/header.js (hunk 1 above)
Comment (garnich) [reply src/components/header/header.js]: done

File: src/components/header/header.js (hunk 2 above)
Comment (garnich) [reply src/components/header/header.js]: done

**Pull request merged**